pkg/asset/cluster/tfvars.go

```diff
@@ -819,6 +824,46 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			masterConfigs[i] = m.Spec.ProviderSpec.Value.Object.(*machinev1.PowerVSMachineProviderConfig)
 		}
 
+		client, err := powervsconfig.NewClient()
+		if err != nil {
+			return err
+		}
+		var (
+			vpcRegion, vpcZone string
+		)
+		vpcName := installConfig.Config.PowerVS.VPCName
+		if vpcName != "" {
+			var vpc *vpcv1.VPC
+			vpc, err = client.GetVPCByName(ctx, vpcName)
+			if err != nil {
+				return err
+			}
+			var crnElems = strings.SplitN(*vpc.CRN, ":", 8)
+			vpcRegion = crnElems[5]
+		} else {
+			specified := installConfig.Config.PowerVS.VPCRegion
+			if specified != "" {
+				if powervstypes.ValidateVPCRegion(specified) {
+					vpcRegion = specified
+				} else {
+					return errors.New("unknown VPC region")
+				}
+			} else if vpcRegion, err = powervstypes.VPCRegionForPowerVSRegion(installConfig.Config.PowerVS.Region); err != nil {
+				return err
+			}
+		}
+		if vpcSubnet != "" {
+			var sn *vpcv1.Subnet
+			sn, err = client.GetSubnetByName(ctx, vpcSubnet, vpcRegion)
+			if err != nil {
+				return err
+			}
+			vpcZone = *sn.Zone.Name
+		} else {
+			rand.Seed(time.Now().UnixNano())
```

Comment:
I am not familiar with powervs but looking at this I see some potentials for errors. What if there are other options for sub nets. Should we search for valid subnets and if you still want to use a random, randomly select from the list of valid subnets ?

Reply:
this is the way we've done it for a while. we (@mjturek and I) made a lot of time making sure it wasn't going to cause problems. however, we know it's not ideal not randomly select the vpc zone this way. but it's fine for now :)

Reply:
Not officially documented, but this knowledge of "valid zones" with the number 1 to 3 as suffix per region is gained from internal discussion with the IBM Cloud team. Therefore, this bit is basically allowing user to reuse his/her existing subnet, making sure it exists and deriving the zone from it, if specified, or we just pick one randomly from existing ones for the installer.

Reply:
thanks for the explanations !